File: csharp/Agent.cs
using System;
using System.Collections.Generic;

namespace TormentNexus {
    public class Agent {
        private Dictionary<string, object> _state;
        private HashSet<object> _listeners;
        private List<object> _steeringQueue;
        private List<object> _followUpQueue;

        public Agent() {
            _state = new Dictionary<string, object>();
            _listeners = new HashSet<object>();
            _steeringQueue = new List<object>();
            _followUpQueue = new List<object>();
        }

        public void AddListener(object listener) {
            _listeners.Add(listener);
        }

        public void RemoveListener(object listener) {
            _listeners.Remove(listener);
        }
    }
}

File: rust/src/agent.rs
use std::collections::HashSet;

pub struct Agent {
    state: String,
    listeners: HashSet<String>,
    steering_queue: Vec<String>,
    follow_up_queue: Vec<String>,
}

impl Agent {
    pub fn new() -> Self {
        Agent {
            state: String::new(),
            listeners: HashSet::new(),
            steering_queue: Vec::new(),
            follow_up_queue: Vec::new(),
        }
    }

    pub fn add_listener(&mut self, listener: String) {
        self.listeners.insert(listener);
    }

    pub fn remove_listener(&mut self, listener: &String) {
        self.listeners.remove(listener);
    }
}